src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,104 @@
+pushed to the cloud are also copied to the local cache for easy access.
+
+@section extension_file_handle FILE_HANDLE
+
+Each storage source also implements the \c WT_FILE_HANDLE interface, to access the files
+on the file system. Since the object store is archival, all the write access methods are disabled.
+
+@section extension_log_stat Logging and Statistics
+All of the cloud storage sources have their own logging system implemented. The storage sources
+provides a logger implemntation that redirects the generated logs to the WiredTiger logging system.
+The SDKs also provide their own logging levels and is translated into WiredTiger logging levels.
+
+@section s3_store S3 Storage source
+The S3 store implements the following key functionalities defined by the storage source interface:
+
+- The bucket is expected to be a string argument, which is the bucket name and the
+corresponding region separated by a ';', e.g. "bucket1;ap-southeast-2". AWS requires an access token
+to authenticate programmatic access to S3. The access token comprises an access key ID and a
+secret access key as a set. The S3 store requires the token to be provided as a single string
+argument of the access key followed by the secret key, separated by a ';'.
+
+- The S3 store also configures a directory on the file system local to the WiredTiger database as a
+cache for the object files. The cache is used to improve the read latency for the object files read
+after an initial download from the S3.
+
+- When A S3 file handle is created, the local cache gets a copy of the S3 object if
+it doesn't already have it. Since the local cache is a directory on the WiredTiger database's file
+system, the file is accessed using a \c WT_FILE_HANDLE implementation native to WiredTiger.
+Hence, a copy of native \c WT_FILE_HANDLE is kept encapsulated in the S3 file handle.
+
+- The S3 store has created a logging class which extends the AWS SDK's logging system.
+
+@section azure_gcp_store Azure and GCP Storage Source
+The Azure and GCP store implements the following key functionalities differently from S3:
+
+- The GCP authentication requires setting a GOOGLE_APPLICATION_CREDENTIALS environment variable 
+which refers to the path of authentication file and the Azure authenication is requires setting an
+AZURE_STORAGE_CONNECTION_STRING environment variable to be set.
+
+- A reference count is incremented each time the file handle reference is opened and decremented
+with the close. When the reference count goes zero, the file system is de-registered from the
+storage source and any allocated resources are freed.
+
+- Whenever a GCP store and Azure file handle is created, the expectation is that the file does not
+exist in the local file system. All the required file handle functions will directly interface with
+the connection class accessing file operations via the SDKs.
+
+- The GCP store has created a logging class which extends the GCP SDK's logging backend and the
```

Comment:
I briefly cover it here because each SDK has their own way of creating the logging system, and the idea here is to specifically mention those. What are your thoughts @ajmorton ?

Reply:
That's a fair point. To me this is an implementation detail that we could drop in favour of the more general "WiredTiger wraps the logs from each store" but this is very much a nit. Ping @korteland for a second opinion but also happy to leave as is.

Reply:
I'm fine with covering it, we have a couple of arch guide pages that get right into the weeds (e.g. fast truncate and RTS).

Reply:
Easy 👍. Convo resolved